resources/js/chat/join-channels.tsx

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import { Spinner } from 'components/spinner';
+import ChannelJson from 'interfaces/chat/channel-json';
+import { computed, makeObservable } from 'mobx';
+import { observer } from 'mobx-react';
+import core from 'osu-core-singleton';
+import * as React from 'react';
+import { classWithModifiers } from 'utils/css';
+import { trans } from 'utils/lang';
+
+type JoinedStatus = 'joined' | 'joining' | null;
+type Props = Record<string, never>;
+
+interface ChannelProps {
+  channel: ChannelJson;
+  onClick: (channelId: number) => void;
+  status: JoinedStatus;
+}
+
+function Channel({ channel, onClick, status }: ChannelProps) {
+  const handleClick = React.useCallback(
+    () => onClick(channel.channel_id),
+    [channel.channel_id, onClick],
+  );
+
+  let statusElement: React.ReactNode | undefined;
+  if (status === 'joined') {
+    statusElement = <i className='fas fa-check' />;
+  } else if (status === 'joining') {
+    statusElement = <Spinner />;
+  }
+
+  return (
+    // anchor instead of button due to Firefox having an issue with button padding in subgrid.
+    <a className={classWithModifiers('chat-join-channel__channel', { joined: status === 'joined' })} onClick={handleClick}>
```

Comment:
anchor without href isn't all that different from just using span... (most notably no tabstop, also wrong cursor)